luigi/scheduler.py

```diff
@@ -546,7 +546,8 @@ def add_task(self, worker, task_id, status=PENDING, runnable=True,
         * add additional workers/stakeholders
         * update priority when needed
         """
-        self.update(worker)
+        worker_id = kwargs['worker']
```

Comment:
Well, I thought that it's "enough" that the first statement directly does access that attribute. Would it be OK to explicitly state that 'worker' must be passed as a kwarg?